### PR TITLE
Tiny enhancements to README_OSX.md

### PR DESCRIPTION
**Pull request by bagong**
- you probably added --configuration Debug by accident, it's the default

As I was at it I added a few cleanups that should be pretty non-controversial. As symlinking and standalone are broken (ever since the switch to Qt, I think) I removed them.

### CODE REVIEW

**Comment by scztt** [comment]
Looks good - except:

I added the "--configuration Debug" because it makes it explicitly clear which config you're building, as well as making clear how you would change that if you wanted. I chose Debug versus Release because, for someone that's building for the first time or isn't used to the build process, debugging a Release build is at best a pain, and at worst a complete non-starter. Making sure someone who just built can set a breakpoint, and step through some code seems more important than them having fully performant build.

**Comment by bagong** [comment]
Ah, okle ;)

**Comment by crucialfelix** [comment]
I agree with keeping the explicit --configuration there so they know how to
change it.

but I would say that default should be Release. the first timer is just
building because they want the latest version 3.7

I have seen hundreds of threads on the list about "why is this thing so
slow", "how do I build a release version" etc.

never has a newbie asked why they can't set a breakpoint.

**Comment by danstowell** [comment]
don't you want --config RelWithDebInfo?

**Comment by bagong** [comment]
Maybe using 'RelWithDebInfo' and explaining a bit is good midway? For example:

**Note:** In the last line cmake starts the build by calling the Xcode toolchain. By default, Xcode builds will be Debug. The example code prevents this by adding `--config RelWithDebInfo`, which results in a faster binary but still provides some debug information. The four possible build configs are Debug, RelWithDebInfo, Release, and MinSizeRel. The path statement after --build `.` points to the file CMakeCache.txt in the build folder, which contains the build configuration settings as generated by cmake.

**Comment by crucialfelix** [comment]
That sounds excellent to me. I would wait for further opinions though.

File: README_OS_X.md [hunk 1]
```diff
@@ -92,11 +91,11 @@ Once those dependencies are satisfied, the following steps will build Supercolli
 	$>	cd supercollider
 	$>	mkdir build; cd build
 	$>	cmake -G Xcode -DCMAKE_PREFIX_PATH=`brew --prefix qt5`  ..
-  $>  cmake --build . --target install --config Debug
+	$>	cmake --build . --target install --config Debug
```
Comment (crucialfelix) [inline README_OS_X.md]: could we change this to RelWithDebInfo ?  its very likely that this example gets copypasted.

then its ready for merge I think

**Comment by bagong** [comment]
Let's have a word from Scott C. It's didactics against pragmatism. If Scott is willing to answer all the emails asking "Why is SC so slow now? Eh, and why does it suddenly say debug build in the post Window" ;)

**Comment by bagong** [comment]
Since I fell in that trap several times now, I'd like to add a consideration: the native build tools (at least Visual Studio and Xcode) seem to default to 'Debug' without caring what was set in the cmake configuration. So it can easily happen that you configure "RelWithDebInfo" in cmake (without actually knowing it, just because it's the default) and Debug as the default setting of the native build tool. At least on Windows there seems to be no mechanism in place that retroactively correct's the cmake setting when calling the native build-tool with it's own build type. On Windows the build still succeeds, but SC behaves very badly then. I am tempted to ask for having cmake default to Debug too :).

**Comment by scztt** [comment]
There are bad scenarios either way. If you build release without knowing and try to set a breakpoint, you'll *probably* hit it - but you won't get variable introspection, and step in/out/over will behave erratically. I still, as an experienced developer, get bit in the ass by this (why the hell can't I step into this function? this doesn't look like the right stack? oh, whoops, this is a release build...). Defaulting to release is a hidden gotcha for people trying to do development work in SC, and defaulting to debug is a hidden gotcha for people trying to do music work in SC.... Both are bad, so I'm agnostic - as long as the readme refers to an EXPLICIT build type (rather than falling back to a default), then I think we're okay.

Regarding the Cmake default: why have a Cmake default at all - if a build type isn't specified, what if we show a detailed error message listing the build types, which one to use, and the per-platform consequences (i.e. with -G Xcode you can change them after the fact, which is not true with some other toolchains). E.g. there are almost no cases where you would want to build with Release vs RelWithDebInfo, but that's not obvious if you're unfamiliar with the cmake paradigms.

**Comment by bagong** [comment]
I read up a bit on the "cmake build type default". Apparently Xcode- and VS- have multi-configuration generators, MinGW and Linux have single configuration generators. With multi-configuration generators any cmake statements that depend on CMAKE_BUILD_TYPE are not evaluated, the build type is only determined by the --config argument in the cmake --build command.
On single configuration generators it is the opposite: only the CMAKE_BUILD_TYPE setting is relevant, any --config value is ignored.
So it is not correct what I said, that "changing" the configuration after the fact can cause problems. There is no fact for VS and XCode and there is no change for MinGW ;)

**Comment by crucialfelix** [comment]
a developer who knows how to set a breakpoint getting bitten in the ass is rarer and more able to realize what he/she did wrong.  

most people are compiling to use it.

**Comment by bagong** [comment]
I agree with Felix
But then should it be "Release" or "RelWithDebInfo" on OS X/Xcode? Look for example at server/plugins/CMakeLists.txt, line 266. I wander what that would mean for RelWithDebInfo. Maybe @timblechmann has a comment here?

**Comment by crucialfelix** [comment]
I'm pretty sure RelWithDebInfo is best and correct. @bagong the branch is yours, can you push that ?

**Comment by bagong** [comment]
Okay, as a practice in braveness :) (@Chris parenthetically: there seems to be a sudden burst in activity. As to Windows, it looks good at the moment, but requiring Qt5.3.2 creates a problem for my build system which uses a rolling release model. Will look at the Quarks when I worked something out for MinGW, hopefully today)

**Pull request merged**